### PR TITLE
cpp-ipfs-api: 2017-01-04 -> 2022-01-30, unbreak build, rename to cpp-ipfs-client-api

### PR DESCRIPTION
**Pull request by wamserma**
ZHF: https://github.com/NixOS/nixpkgs/issues/172160

###### Description of changes

- bumped version
- unvendored lohmann-json
- fixed build
- adopted upstream name change (with alias for backward-compatibility)

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by risicle** [comment]
`nixpkgs-review` happy, macos 10.15 & nixos x86_64.

Broken on aarch64 but that's not new.

**Comment by wamserma** [comment]
Result of `nixpkgs-review pr 173197` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>cpp-ipfs-http-client</li>
  </ul>
</details>

**Comment by wamserma** [comment]
> `nixpkgs-review` happy, macos 10.15 & nixos x86_64.
> 
> Broken on aarch64 but that's not new.

updated the platforms attribute

File: pkgs/development/libraries/cpp-ipfs-http-client/default.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    repo = "cpp-ipfs-http-client";
+    rev = "3cdfa7fc6326e49fc81b3c7ca43ce83bdccef6d9";
+    sha256 = "sha256-/oyafnk4SbrvoCh90wkZXNBjknMKA6EEUoEGo/amLUo=";
+  };
+
+  patches = [ ./unvendor-nlohmann-json.patch ];
+
+  postPatch = ''
+  substituteInPlace CMakeLists.txt \
+      --replace '# Fetch "JSON for Modern C++"' "include_directories(${nlohmann_json}/include)"
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/cpp-ipfs-http-client/default.nix:18]:
```suggestion
    substituteInPlace CMakeLists.txt \
      --replace '# Fetch "JSON for Modern C++"' "include_directories(${nlohmann_json}/include)"
```

File: pkgs/development/libraries/cpp-ipfs-http-client/default.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
+{ lib, stdenv, fetchFromGitHub, curl, cmake, nlohmann_json }:
+
+stdenv.mkDerivation {
+  pname = "cpp-ipfs-http-client";
+  version = "2022-01-30";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/cpp-ipfs-http-client/default.nix:5]:
```suggestion
  version = "unstable-2022-01-30";
```

**Comment by SuperSandro2000** [comment]
@ofborg build cpp-ipfs-http-client

**Pull request merged**